vertx-web/src/main/java/io/vertx/ext/web/handler/sse/impl/EventSourceImpl.java

```diff
@@ -0,0 +1,137 @@
+package io.vertx.ext.web.handler.sse.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxException;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.handler.sse.EventSource;
+import io.vertx.ext.web.handler.sse.SSEHeaders;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EventSourceImpl implements EventSource {
```

Comment:
You need to think about synchronizing access to your fields. As commented elsewhere, this may be used in a worker verticle.